src/riak_kv_vnode.erl

```diff
@@ -134,6 +115,8 @@
           leasing = false :: boolean()
          }).
 
+-type update_hook() :: module() | undefined.
```

Comment:
Thoughts on defaulting to a `riak_kv_noop_update_hook` module, rather than `undefined` here? Implement the no-op versions of all the functions and we can get rid of all of the function heads that check for `undefined` and just call the noop version... and change all the `maybe_*` functions to just `*`.